runners/flink/src/test/java/org/apache/beam/runners/flink/translation/FlinkPipelineTranslatorUtilsTest.java

```diff
@@ -0,0 +1,78 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.translation;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.Serializable;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.runners.core.construction.PipelineTranslation;
+import org.apache.beam.runners.core.construction.graph.ExecutableStage;
+import org.apache.beam.runners.core.construction.graph.GreedyPipelineFuser;
+import org.apache.beam.runners.flink.translation.utils.FlinkPipelineTranslatorUtils;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.Impulse;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.junit.Test;
+
+/**
+ * Tests for {@link org.apache.beam.runners.flink.translation.utils.FlinkPipelineTranslatorUtils}.
+ */
+public class FlinkPipelineTranslatorUtilsTest implements Serializable {
+
+  @Test
+  /* Test for generating readable operator names during translation. */
+  public void testOperatorNameGeneration() throws Exception {
+    Pipeline p = Pipeline.create();
+    p.apply(Impulse.create())
+        // Anonymous ParDo
+        .apply(
+            ParDo.of(
+                new DoFn<byte[], String>() {
+                  @ProcessElement
+                  public void processElement(
+                      ProcessContext processContext, OutputReceiver<String> outputReceiver) {}
+                }))
+        // Name ParDo
+        .apply(
+            "MyName",
+            ParDo.of(
+                new DoFn<String, Integer>() {
+                  @ProcessElement
+                  public void processElement(
+                      ProcessContext processContext, OutputReceiver<Integer> outputReceiver) {}
+                }));
+
+    ExecutableStage firstEnvStage =
+        GreedyPipelineFuser.fuse(PipelineTranslation.toProto(p))
+            .getFusedStages()
+            .stream()
+            .findFirst()
+            .get();
+    RunnerApi.ExecutableStagePayload basePayload =
+        RunnerApi.ExecutableStagePayload.parseFrom(
+            firstEnvStage.toPTransform("foo").getSpec().getPayload());
+
+    String executableStageName =
+        FlinkPipelineTranslatorUtils.genOperatorNameFromStagePayload(basePayload);
+
+    assertThat(executableStageName, is("ExecutableStage(2)[ParDo(Anonymous)][MyName]"));
```

Comment:
Can we skip or shorten `ExecutableStage`? In the metric backend system characters usually come at a premium and it does not really add anything valuable for the user.

Reply:
I think we could skip the word `ExecutableStage` entirely. Maybe just enclosing it in `{}` would suffice to indicate that a harness is involved.

I realized the names are less readable for Python. In Java we would just get the user-defined name (if set).